### PR TITLE
Document the Process Console 

### PR DESCRIPTION
**Pull request by JADarius**
### Pull Request Overview

This pull request adds documentation for the `process_console` capsule so that it can be easier for new users to get started. I also found a typo in the comments of `process_console.rs`.

### Testing Strategy

N/A

### TODO or Help Wanted

This pull request still needs a review and suggestions for possible improvement.

### Documentation Updated

- [x] Added the relevant file in `/doc`.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/process_console.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 //! --------
 //!
 //! This module provides a simple text-based console to inspect and control
-//! which processes are running. The console has five commands:
+//! which processes are running. The console has eleven commands:
```
Comment (bradjc) [inline capsules/src/process_console.rs:8]: It's probably unlikely we'll be able to keep these comments in sync with the new document. I would rather just remove the duplicates from here and instead direct readers to the new doc file.

File: doc/Process_Console.md [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+   restarts.
+ - `Name`: The process name.
+ - `Quanta`: How many times this process has exceeded its allotted time
+   quanta.
+ - `Syscalls`: The number of system calls the process has made to the kernel.
+ - `Restarts`: How many times this process has crashed and been restarted by
+   the kernel.
+ - `Grants`: The number of grants that have been initialized for the process
+   out of the total number of grants defined by the kernel.
+ - `State`: The state the process is in.
```
Comment (bradjc) [inline doc/Process_Console.md:36]: Move this below to the example of the list command.

File: doc/Process_Console.md [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+  - `status` - prints the current system status
+  - `list` - lists the current processes with their IDs and running state
+  - `stop n` - stops the process with name n
+  - `start n` - starts the stopped process with name n
+  - `fault n` - forces the process with name n into a fault state
+  - `terminate n` - terminates the running process with name n, moving it to the Terminated state
+  - `boot n` - tries to restart a Terminated process with name n
+  - `panic` - causes the kernel to run the panic handler
+  - `kernel` - prints the kernel memory map
+  - `process n` - prints the memory map of process with name n
```
Comment (bradjc) [inline doc/Process_Console.md:22]: Move this below as a sort of table of contents for the more in depth descriptions and examples of each command.

File: doc/Process_Console.md [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+                  115200,
+                  &mut console::WRITE_BUF,
+                  &mut console::READ_BUF,
+                  &mut console::COMMAND_BUF,
+                  kernel,
+                  Capability));
+ hil::uart::UART::set_client(&usart::USART0, pconsole);
+
+ pconsole.start();
+ ```
```
Comment (bradjc) [inline doc/Process_Console.md:63]: Having a setup section is ok, but it needs to be updated with an example of using the component. I doubt this code would even work anymore.

File: doc/Process_Console.md [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+tock$ help
+Welcome to the process console.
+Valid commands are: help status list stop start fault boot terminate process kernel panic
+ ```
+
+In the following paragraphs the outputs of the listed commands will be shown:
+
+ - To see all of the processes on the board, use `list`:
+
+ ```text
```
Comment (bradjc) [inline doc/Process_Console.md:105]: All of these code blocks here down need to be indented so they render correctly as part of the list.

**Review by bradjc — COMMENTED** [review]
Cool! Also please add a link from doc/README.md.

File: doc/Process_Console.md (hunk 4 above)
Comment (alexandruradovici) [reply doc/Process_Console.md:63]: You can take a look at how microbit instantiates it.

Comment (JADarius) [reply doc/Process_Console.md]: I updated the setup, but I don't know if I did it correctly.

File: doc/Process_Console.md [hunk 6]
```diff
@@ -0,0 +1,595 @@
...
+    )
+    .finalize(components::uart_mux_component_static!()),
+     components::alarm::AlarmMuxComponent::new(rtc)
+        .finalize(components::alarm_mux_component_static!(nrf52::rtc::Rtc)),
+     process_printer,
+ )
+ .finalize(components::process_console_component_static!(
+        nrf52833::rtc::Rtc
+ ));
+ let _ = _process_console.start();
```
Comment (bradjc) [inline doc/Process_Console.md:37]: Can we just include the simplified version of this? That is, assume we already have `uart_mux` and `alarm_mux` objects?

File: doc/Process_Console.md [hunk 7]
```diff
@@ -0,0 +1,595 @@
+ Process Console
+ ===============
+ 
+ `process_console` is a capsule that implements a small shell over the UART that allows
+ a terminal to inspect the kernel and control userspace processes.
+
+  - [Setup](#setup)
+  - [Using Process Console](#using-processconsole)
+  - [Commands](#commands)
```
Comment (bradjc) [inline doc/Process_Console.md:9]: Please wrap this in the ToC comments (see another doc/ markdown file for an example).

File: capsules/src/process_console.rs [hunk 8]
```diff
@@ -5,7 +5,7 @@
...
 //!  - 'help' prints the available commands and arguments
```
Comment (bradjc) [inline capsules/src/process_console.rs:9]: Can you delete from this file what you moved to the new document?

**Comment by JADarius** [comment]
I have made the requested changes.

File: capsules/src/process_console.rs [hunk 9]
```diff
@@ -1,111 +1,3 @@
-//! Implements a text console over the UART that allows
-//! a terminal to inspect and control userspace processes.
```
Comment (bradjc) [inline capsules/src/process_console.rs:2]: Leave this and add empty line then a link to new doc.

**Review by alistair23 — APPROVED** [review]

**Review by alexandruradovici — APPROVED** [review]

**Pull request merged**